From b2da82c636465ac27be2f1dc6b33321761d17b0c Mon Sep 17 00:00:00 2001
From: Ted Ross <tross@apache.org>
Date: Thu, 24 Jun 2010 18:48:53 +0000
Subject: [PATCH 10/11] Bug 607754 - 1.3 qpid-config does not work against 1.2 qpidd

Handle the case where older versions of the broker supply no delivery-properties to messages
originated from within the broker.

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@957676 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit a5ec9d6b06be38e71a776d82f11c867e0dc4f003)
---
 qpid/extras/qmf/src/py/qmf/console.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qpid/extras/qmf/src/py/qmf/console.py b/qpid/extras/qmf/src/py/qmf/console.py
index af7a895..062cb6f 100644
--- a/qpid/extras/qmf/src/py/qmf/console.py
+++ b/qpid/extras/qmf/src/py/qmf/console.py
@@ -2381,7 +2381,7 @@ class Broker(Thread):
       #
       dp = msg.get("delivery_properties")
       rkey = None
-      if dp.routing_key:
+      if dp and dp.routing_key:
         rkey = dp.routing_key
         items = rkey.split('.')
         if len(items) >= 4:
-- 
1.7.0.1

